smse/pipelines/image.py

```diff
@@ -22,43 +37,66 @@ class ImagePipeline(BasePipeline):
     def __init__(self, config: ImageConfig):
         super().__init__(config)
         self.config: ImageConfig = config
+        # Initialize the transform Pipeline.
+        self.transform_pipeline = self._create_transform_pipeline()
+
+    def _create_transform_pipeline(self) -> transforms.Compose:
+        """Create a transformation pipeline for image preprocessing."""
+        transform_list = [
+            transforms.Lambda(self._ensure_rgb),  # Ensure RGB Color mode.
+            transforms.ToTensor(),  # Convert PIL to Tensor
+        ]
+
+        if self.config.target_size:  # Resize images.
+            transform_list.append(transforms.Resize(self.config.target_size))
+
+        if self.config.normalize:  # Normalize images.
+            transform_list.append(transforms.Normalize(mean=self.config.mean, std=self.config.std))
+
+        return transforms.Compose(transform_list)
+
+    @staticmethod
+    def _ensure_rgb(image: Image.Image) -> Image.Image:
+        """Ensure image is in RGB mode."""
+        if image.mode != "RGB":
+            image = image.convert("RGB")
+        return image
```

Comment:
This is hard coded for RGB, right? what if the user needs that images in BGR? 

You should also use native torchvision transforms instead of you custom one. I believe we can use [this transform](https://pytorch.org/vision/main/generated/torchvision.transforms.v2.functional.permute_channels.html?highlight=bgr) to handle the conversion between different modes

```
>>> rgb_image = torch.rand(3, 256, 256)
>>> bgr_image = F.permute_channels(rgb_image, permutation=[2, 1, 0])
```

We can define a set of permutations for each mode and apply the correct one based on the user's input.

There are also these transforms that might be handy
- RGB: https://pytorch.org/vision/main/generated/torchvision.transforms.v2.RGB.html?highlight=rgb#torchvision.transforms.v2.RGBhttps://pytorch.org/vision/main/generated/torchvision.transforms.v2.RGB.html?highlight=rgb#torchvision.transforms.v2.RGB
- GrayScale: https://pytorch.org/vision/main/generated/torchvision.transforms.v2.Grayscale.html?highlight=grayscale#torchvision.transforms.v2.Grayscale